datalad/crawler/pipelines/balsa.py

```diff
@@ -0,0 +1,219 @@
+                self.repo.remove(item)
+                files = list(f_f('.*', topdir='_files'))
+            else:
+                lgr.warning("%s does not exist in the individaully listed files by name, "
+                            "but will be kept from canconical tarball" % item)
+        if files:
+            lgr.warning("The following files do not exist in the canonical tarball, but are "
+                        "individually listed files and will not be kept: %s" % files)
+            rmtree(files_path)
+
+        yield data
+
+
+def pipeline(dataset_id, url=TOPURL):
+    lgr.info("Creating a pipeline for the BALSA dataset %s" % dataset_id)
+    annex = Annexificator(create=False,
+                          statusdb='json',
+                          special_remotes=[ARCHIVES_SPECIAL_REMOTE, DATALAD_SPECIAL_REMOTE],
+                          options=["-c",
+                                   "annex.largefiles="
+                                   "exclude=Makefile and exclude=LICENSE* and exclude=ISSUES*"
+                                   " and exclude=CHANGES* and exclude=README*"
+                                   " and exclude=*.[mc] and exclude=dataset*.json"
+                                   " and exclude=*.txt"
+                                   " and exclude=*.json"
+                                   " and exclude=*.tsv"
+                                   ])
+
+    if not exists("_files"):
+        makedirs("_files")
+
+    def splitpath(data):
+        data = data.copy()
+        fullpath = data.pop('path')
+        path = os.path.dirname(fullpath)
+        if path:
+            data['path'] = path
+        data['filename'] = os.path.basename(fullpath)
+        yield data
+
+    files_url = opj(url, 'file/show/')  # files_url = https://balsa.wustl.edu/file/show/
+
+    url = opj(url, 'study/')  # url = https://balsa.wustl.edu/study/
+    dataset_url = '%sshow/%s' % (url, dataset_id)  # url = https://balsa.wustl.edu/study/show/[dataset_id]
+
+    balsa = BalsaSupport(repo=annex.repo)
+
+    # BALSA has no versioning atm, so no changelog either
+    return [
+        annex.switch_branch('incoming'),
+        [
+            crawl_url(url),
+            [
+                assign({'dataset': dataset_id}),
+                skip_if({'dataset': 'test study upload'}, re=True),
+            ],
+            [
+                crawl_url(dataset_url),
+                [
+                    extract_meta,
+                    annex,
+                ],
+                [
+                    # canonical tarball
+                    a_href_match('.*/download/.*', min_count=1),
+                    annex,
+                ],
+                [
+                    a_href_match(files_url, min_count=2),
+                    assign({'path': '_files/%(url_text)s'}, interpolate=True),
+                    sub({'path': {' / ': '/'}}),
+                    splitpath,
+                    annex,
```

Comment:
so here it would fetch the 'show' page for that file, not the actual file from "Download" link on that show page.... Ideally we want to fetch that target url, not just the page.  Then we would get direct link to that file instead of just a url "from a tarball" ... or am I misunderstanding something?  also since I didn't see the warning that content of this _files/ differs from the actual file from tarball, I suspect that smth is not working as it should
BUT anyways -- there are tricky cases, e.g. for the https://balsa.wustl.edu/sceneFile/show/1vrP  (for .scene file) it points to another .zip (Glasser_et_al_2016_HCP_MMP1.0_RVVG.zip) which would contain that scene and then some other (probably referenced within the scene) .gii files.  I could also swear that while 'show/'ing some other file I got for it to have a Download of .zip of much bigger size (650MB) which had the same filename but obviously different content... found it:  https://balsa.wustl.edu/sceneFile/show/lLMz    will report "upstream" now